flask/app/analyses.py

```diff
@@ -303,6 +315,12 @@ def create_analysis():
         updated_by_id=updated_by_id,
         datasets=found_datasets,
     )
+
+    enum_error = mixin(analysis, request.json, ["priority"])
```

Comment:
Overall the backend looks great and works for me! Only issue I came across during testing was that in the POST request, when an invalid enum for priority is supplied in the request, an error 400 is thrown but after looking into this, it's not because of `enum_error` (the priority field is actually successfully updated) but because it appears that the instance is failing to be added to the database.
```
curl -i -X POST \
    -H "Content-Type: application/json" \
    --data '{"datasets" : ["1"],
            "pipeline_id" : 1,
            "priority" : "Non"}'  localhost:5000/api/analyses
HTTP/1.0 400 BAD REQUEST
Content-Type: application/json
Content-Length: 63
Server: Werkzeug/1.0.1 Python/3.7.9
Date: Wed, 21 Apr 2021 12:43:43 GMT

{
  "error": "Data truncated for column 'priority' at row 1"
}
```
The correct function should be `enum_validate` (for this function the first argument should be the the model itself rather than an instance of the model). The two functions do look alike and I'll admit are not named very well! I believe the difference is that `mixin` is used for allowing only specific fields to be edited, whereas `enum_validate` is for determining whether an instance of a model to be added to the database has valid enums. It also looks like some logic to account for 'None' will also need to be added like what you did for the PATCH endpoint. The logic for enum validation could also be moved higher up to say before the query so the work from the query is not wasted if priority is not a valid enum.

Reply:
Nice catch. Updated in [5b4fd5b](https://github.com/ccmbioinfo/stager/pull/522/commits/5b4fd5bef2a15a6f722cb30e842a448f9cf522ef) and [7ff4505](https://github.com/ccmbioinfo/stager/pull/522/commits/7ff4505207db53a84ebefd3aa4850a6a3bae5892). For the "None" value, I'm not sure we'll need it in the POST route? My reasoning is that when POSTing, json null would be sufficient b/c the column is nullable, so just passing nothing or null should take care of that. It would probably be better to pass `null` on the PATCH route too (rather than `"None"`) -- that might require updating the `mixin` function to check whether the column is nullable and allow null values to pass through.